src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php

```diff
@@ -37,7 +38,14 @@ class JsonManifestVersionStrategy implements VersionStrategyInterface
     public function __construct(string $manifestPath, HttpClientInterface $httpClient = null)
     {
         $this->manifestPath = $manifestPath;
-        $this->httpClient = $httpClient;
+
+        if (0 === strpos(parse_url($this->manifestPath, \PHP_URL_SCHEME), 'http')) {
+            if (null === $this->httpClient && !class_exists(HttpClient::class)) {
+                throw new \LogicException(sprintf('The "%s" class requires the "HttpClient" component. Try running "composer require symfony/http-client".', self::class));
+            }
+
+            $this->httpClient = $httpClient ?? HttpClient::create();
```

Comment:
this was rejected in the original PR: we want this to be 100% explicit, so we want that ppl *have to* inject a client if they want to accept HTTP json.
Can you please add a test case about this?